miio/interfaces/vacuum.py

```diff
@@ -0,0 +1,32 @@
+from typing import Dict, Protocol, runtime_checkable
+
+from miio import DeviceStatus
+
+FanspeedPresets = Dict[str, int]
+
+
+@runtime_checkable
+class Vacuum(Protocol):
+    def start(self):
+        """Start vacuuming."""
+
+    def pause(self):
+        """Pause vacuuming."""
```

Comment:
Some implementation does not contain pause. It would be good to describe, what should happen, if the method is not supported. Raise an exception?

Reply:
On roborock, calling pause differs from stop in sense that resuming from a pause will continue using the existing internal map if the cleaning is resumed, in contrast to starting a completely new cleaning round.